src/components/ShoppingControls/ShoppingControls.js

```diff
@@ -0,0 +1,81 @@
+import React, { useContext, useState } from "react";
+import { ShoppingCartContext } from "../../utils/context/ShoppingCartContext";
+import { ShoppingStateEnums } from "../../utils/reducers/ShoppingCartReducer";
+import ReactModal from "react-modal";
+import { Link } from "react-router-dom";
+
+const ShoppingControls = ({ cartItem }) => {
+  const [quantity, setQuantity] = useState(0);
+  const [shoppingCartState, dispatchShoppingCart] =
+    useContext(ShoppingCartContext);
+  const isAddBtnDisabled = quantity == 0;
+  const [modalIsOpen, setModalIsOpen] = React.useState(false);
+  const addBtnCartTitle = isAddBtnDisabled
+    ? "Please select a quantity"
+    : "Add to Cart";
+  ReactModal.setAppElement("#app");
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    dispatchShoppingCart({
+      type: ShoppingStateEnums.ADD_ITEM,
+      payload: { ...cartItem, quantity: parseInt(quantity)},
+    });
+
+    setModalIsOpen(!modalIsOpen);
+    setQuantity(0);
+  };
+
+  return (
+    <div className="shopping-controls">
+      <form onSubmit={handleSubmit}>
```

Comment:
My recommendation here is to try to find an e-commerce site with a similar approach to the selects or buttons in order to add products to the cart. The reason is to see how they implemented it or how it works, not sure if adding a form for each product is the correct, also you are hiding/showing the submit button, which is accessibility terms that could be not correct because that kind of users are not aware of that effect.

Reply:
done